app/bundles/EmailBundle/Controller/EmailController.php

```diff
@@ -1637,4 +1670,102 @@ protected function getDefaultOrderDirection(): string
     {
         return 'DESC';
     }
+
+    protected function getEmailModel(): EmailModel
+    {
+        $model = $this->getModel('email');
+        \assert($model instanceof EmailModel);
+
+        return $model;
+    }
+
+    /**
+     * @return array<int|string, array<int|string, int|string>>
+     *
+     * @throws Exception
+     */
+    private function getCountriesTableData(Email $entity): array
+    {
+        $model = $this->getEmailModel();
+
+        // get A/B test information
+        $parent = $entity->getVariantParent();
+
+        // get translation parent
+        $translationParent = $entity->getTranslationParent();
+        $includeVariants   = (($entity->isVariant() && empty($parent)) || ($entity->isTranslation() && empty($translationParent)));
+
+        return $model->getCountryStats(
+            $entity,
+            $includeVariants,
+        );
+    }
+
+    /**
+     * @return array<int, string>
+     */
+    private function getCountriesTableExportHeader(): array
+    {
+        return [
+            $this->translator->trans('mautic.lead.lead.thead.country'),
+            $this->translator->trans('mautic.email.graph.line.stats.sent'),
+            $this->translator->trans('mautic.email.graph.line.stats.read'),
+            $this->translator->trans('mautic.email.clicked'),
+        ];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function countryStatsAction(
+        int $objectId
+    ): Response {
+        $model  = $this->getEmailModel();
+        $entity = $model->getEntity($objectId);
+
+        if (empty($entity) || !$this->security->hasEntityAccess(
+            'email:emails:viewown',
+            'email:emails:viewother',
+            $entity->getCreatedBy()
+        )) {
+            throw new AccessDeniedHttpException();
+        }
+
+        $statsCountries = $this->getCountriesTableData($entity);
+
+        return $this->render(
+            '@MauticCore/Helper/countries_table.html.twig',
+            [
+                'data'           => $statsCountries,
+                'object'         => $entity,
+            ]
+        );
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function exportCountriesStatsAction(int $objectId, string $format = 'csv'): StreamedResponse|Response
```

Comment:
```suggestion
    public function exportCountriesStatsAction(EmailModel $model, ExportHelper $exportHelper, int $objectId, string $format = 'csv'): StreamedResponse|Response
```
Same for this method, of course you will also need to adjust the references in the code.